iFixit/react-commerce
# Product page: start adding tests

## Background
Previously, we weren't setting the `NODE_ENV` properly in the `main.yml` workflow, therefore, it was using the `frontend/.env.production` ([example](https://github.com/iFixit/react-commerce/actions/runs/3107844588/jobs/5036395179#step:6:24)) instead of `frontend/.env.test`.
## Summary of changes
In this pull:
1. Now we're using `frontend/.env.test` to run cypress test in E2E workflow ([example](https://github.com/iFixit/react-commerce/actions/runs/3108801290/jobs/5038391640#step:6:24)).
2. Turned on `FLAG__PRODUCT_PAGE_ENABLED` to true in `.env.test` to start testing product pages. 
3. Unskipped a test that was skipped which tests clicking on `View` button from `/Parts` page and verifies if the view button link matches the clicked product page link. 

## CR notes
In `frontend/.env.test` I had to switch `NEXT_PUBLIC_DEFAULT_STORE_CODE`  from `us` to `test` because I was getting the `Store not found` error:  here's an [example](https://github.com/iFixit/react-commerce/actions/runs/3108295882/jobs/5037334565#step:6:112). 

However, there is an issue with using `us` as `STORE_CODE` because all products become `out-of-stock` when running the app locally. To resolve `out-of-stock` issue locally, we can duplicate the `us` store and rename it to `test` with the correct credentials, [more details here](https://ifixit.slack.com/archives/C01K32B4FPX/p1663260330576119?thread_ts=1663259552.812549&cid=C01K32B4FPX). However, I don't think we can do manually duplicate it in CI when the workflow is running. 

Question: @dhmacs which store do we ideally want to use for `NEXT_PUBLIC_DEFAULT_STORE_CODE` in `.env.test` when running the cypress tests in Github Actions? If it's the `test` store, is it possible to add the `test` store in production so it would be copied to local dev?

## QA notes
Please make sure the `E2E tests` ci check is passing.
Also, make sure the previews are still working. 


Connects #452
